core/src/main/java/org/apache/druid/collections/CombiningFunction.java

```diff
@@ -17,11 +17,10 @@
  * under the License.
  */
 
-package org.apache.druid.java.util.common.guava.nary;
+package org.apache.druid.collections;
 
-/**
- */
-public interface BinaryFn<Type1, Type2, OutType>
+@FunctionalInterface
+public interface CombiningFunction<T>
```

Comment:
Ah, good point. All of our custom functions gave me tunnel vision and I forgot that Java exists 😅.  I don't think the existence of `CombiningFunction` is really justified since it's just for vanity/cosmetic reasons to clarify what it's doing, which I don't think are necessary.

Reply:
I have removed this interface with the latest commit to this PR.